daphne_server/src/router/mod.rs

```diff
@@ -0,0 +1,455 @@
+        let status = if matches!(error, DapError::Fatal(_)) {
+            // self.error_reporter.report_abort(&e);
+            StatusCode::INTERNAL_SERVER_ERROR
+        } else {
+            StatusCode::BAD_REQUEST
+        };
+        tracing::error!(error = ?error, "request aborted");
+        let problem_details = error.into_problem_details();
+        metrics
+            .dap_abort_counter
+            // this to string is bounded by the
+            // number of variants in the enum
+            .with_label_values(&[&problem_details.title])
+            .inc();
+        let headers = AppendHeaders([(CONTENT_TYPE, "application/problem+json")]);
+
+        Self((status, headers, Json(problem_details)).into_response())
+    }
+
+    pub fn from_result<E>(result: Result<DapResponse, E>, metrics: &DaphneServiceMetrics) -> Self
+    where
+        E: Into<DapError>,
+    {
+        match result {
+            Ok(o) => Self::new_success(o, metrics),
+            Err(e) => Self::new_error(e, metrics),
+        }
+    }
+}
+
+impl IntoResponse for AxumDapResponse {
+    fn into_response(self) -> axum::response::Response {
+        self.0
+    }
+}
+
+/// An axum extractor capable of parsing a [`DapRequest`].
+#[derive(Debug)]
+struct DapRequestExtractor(pub DapRequest<DaphneAuth>);
+
+#[async_trait]
+impl<S> FromRequest<S, axum::body::Body> for DapRequestExtractor
+where
+    S: Send + Sync,
+{
+    type Rejection = (StatusCode, String);
+
+    async fn from_request(
+        req: Request<axum::body::Body>,
+        state: &S,
+    ) -> Result<Self, Self::Rejection> {
+        #[derive(Debug, Deserialize)]
+        #[serde(deny_unknown_fields)]
+        struct PathParams {
+            version: DapVersion,
+            #[serde(
+                default,
+                deserialize_with = "daphne::messages::base64url::deserialize_opt"
+            )]
+            task_id: Option<TaskId>,
+            #[serde(
+                default,
+                deserialize_with = "daphne::messages::base64url::deserialize_opt"
+            )]
+            agg_job_id: Option<AggregationJobId>,
+            #[serde(
+                default,
+                deserialize_with = "daphne::messages::base64url::deserialize_opt"
+            )]
+            collect_job_id: Option<CollectionJobId>,
+        }
+
+        let (mut parts, body) = req.into_parts();
+        let Path(PathParams {
+            version,
+            task_id,
+            agg_job_id,
+            collect_job_id,
+        }) = Path::from_request_parts(&mut parts, state)
+            .await
+            .map_err(|e| (StatusCode::BAD_REQUEST, e.to_string()))?;
+
+        let sender_auth = DaphneAuth {
+            bearer_token: parts
+                .headers
+                .get("DAP-Auth-Token")
+                .and_then(|v| v.to_str().ok())
+                .map(BearerToken::from),
+            // TODO(mendess) figure out tls
+            cf_tls_client_auth: None,
```

Comment:
How does this trait solve the problem? We just have to parse the certificate and store it in the `cf_tls_client_auth` field.

Reply:
What I'm trying to avoid here is having to re-implement the `verified` string passed from FL. (i.e., "SUCCESS" if a cert was presented and was valid, "FAILED(...)" if there was some sort of error, something else if there was no certificate at all.) Frankly this is some Cloudflare-specific awkwardness that we can and should avoid if we're terminating TLS ourselves.